src/android/toga_android/widgets/textinput.py

```diff
@@ -18,17 +20,22 @@ def set_readonly(self, value):
         self.interface.factory.not_implemented('TextInput.set_readonly()')
 
     def set_placeholder(self, value):
-        # self.native.cell.placeholderString = self._placeholder
-        self.interface.factory.not_implemented('TextInput.set_placeholder()')
+        self.native.setHint(value)
 
     def set_alignment(self, value):
-        self.interface.factory.not_implemented('TextInput.set_alignment()')
+        self.native.setGravity({
+                LEFT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.LEFT,
+                RIGHT_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.RIGHT,
+                CENTER_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
+                JUSTIFIED_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
+                NATURAL_ALIGNED: Gravity.CENTER_VERTICAL | Gravity.CENTER_HORIZONTAL,
+            }[value])
 
     def set_font(self, value):
         self.interface.factory.not_implemented('TextInput.set_font()')
 
     def get_value(self, value):
-        self.interface.factory.not_implemented('TextInput.get_value()')
+        return str(self.native.getText())
```

Comment:
Is the str() cast needed here?

Reply:
Yes, if you try to call len(...getText()), without cast str() before, you got some error like: `TypeError: object of type 'org.python.java.Object' has no len()`, you can test it changing the line https://github.com/jvmaia/celsius_fahrenheit-voc/blob/master/celsius_fahrenheit/app.py#L63